### ISSUE TITLE
Text generation isn't predicting correctly

### ISSUE THREAD

**Issue by limkhaiin1012** [tensorflow/tensorflow#54380]
hello, 

I am building a text generation model in Chinese and trying to pull out the probabilities between words for another task. This is the model I follow : https://www.tensorflow.org/text/tutorials/text_generation.

import tensorflow as tf
from tensorflow.keras import models, layers, preprocessing, Model, callbacks, activations

inputShape = (X_converted_final.shape[1],)
input = layers.Input(shape=inputShape)
x=layers.Embedding(vocab_size2, embedding_dim, input_length=50, trainable=True, embeddings_initializer="uniform")(input)
x = layers.GRU(rnn_units, return_sequences=True, bias_initializer="zeros")(x)
output= layers.Dense(vocab_size2, activation=activations.softmax)(x)

gru_fpi_model = models.Model(input, output)
gru_fpi_model.compile(loss=tf.losses.SparseCategoricalCrossentropy(from_logits=False),
              optimizer='adam', metrics=["accuracy"])
callback = callbacks.EarlyStopping(monitor='accuracy', patience=1)
gru_fpi_model.summary()

So this model should be able to give me '天' if ’昨’ is provided in the sequence, since '昨天' means 'yesterday' in Chinese and we have a lot of examples in the training data. This model always takes a sequence so I tried to give the model a sequence and regard the highest probabilities of next words as the best candidate, given a previous word.  Here is what I did.

prediction=np.argmax(loaded_model(sentence)[:,2,:], axis=-1)

With this line, if I give the model a sentence '我昨天晚上去上学', the model should give me the word that has the highest probabilities after '昨' (in the second position, including the beginning marker of sentence), which should be '天' or other words, but it instead gives me '昨'. I couldn't find any sequence '昨昨' in the training data so there must be something wrong. For some words, it predicts well, e.g. '我', it gives me '們'and it means 'we' in English. So I think this model still works for some words, but it just doesn't work for some.

Could anyone give me some hints on this? Thank you so much

**Comment by mohantym** [comment]
Hi @limkhaiin1012 ! 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!
